addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -354,6 +354,18 @@ export class ProductScreen extends Component {
                 "pos_categ_ids",
                 this.pos.selectedCategory.id
             );
+            if (this.pos.selectedCategory.child_id) {
+                const descendants = this.pos.selectedCategory.getAllChildren(
+                    this.pos.selectedCategory
+                );
+                for (const descendant of descendants) {
+                    const products = this.pos.models["product.product"].getBy(
+                        "pos_categ_ids",
+                        descendant.id
+                    );
+                    list = list.concat(products);
```

Comment:
also, here you might have the same product twice. Can you make sure to remove all duplicates before returning the list?